pkg/services/pluginsintegration/clientmiddleware/tracing_middleware.go

```diff
@@ -0,0 +1,86 @@
+package clientmiddleware
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/codes"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana/pkg/infra/tracing"
+	"github.com/grafana/grafana/pkg/plugins"
+)
+
+// NewTracingMiddleware returns a new middleware that creates a new span on every method call.
+func NewTracingMiddleware(tracer tracing.Tracer) plugins.ClientMiddleware {
+	return plugins.ClientMiddlewareFunc(func(next plugins.Client) plugins.Client {
+		return &TracingMiddleware{
+			tracer: tracer,
+			next:   next,
+		}
+	})
+}
+
+type TracingMiddleware struct {
+	tracer tracing.Tracer
+	next   plugins.Client
+}
+
+func (m *TracingMiddleware) traceWrap(ctx context.Context, opName string) (context.Context, func(error)) {
+	ctx, span := m.tracer.Start(ctx, opName)
+	return ctx, func(err error) {
+		span.End()
```

Comment:
This should probably be deffered since you're calling `SetStatus` and `RecordError` after it's ended, no? Or else just added after the `if` perhaps